src/store/bankdata/getters.ts

```diff
@@ -1,21 +1,39 @@
 import { GetterTree } from 'vuex';
 import { BankAccount } from '@src/view_models/BankAccount';
 import { Validity } from '@src/view_models/Validity';
+import { looksLikeIban } from '@src/util/bank_account_number_helpers';
 
+// It's important to note here that !bankDataIsInvalid is not the same as bankDataIsValid and
+// vice versa. This is because we have states Validity.INCOMPLETE and Validity.RESTORED. If
+// the validity of one of the values is one of these states then both of these will return false.
 export const getters: GetterTree<BankAccount, any> = {
 	bankDataIsInvalid: function ( state: BankAccount ): boolean {
-		return state.validity.bankdata === Validity.INVALID;
+		if ( state.values.accountNumber === '' || looksLikeIban( state.values.accountNumber ) ) {
```

Comment:
I'm still confused here because I thought you clearly separated IBAN+BIC vs accountNumber+bankCode in the state (`index.ts`). Does the `accountNumber` still have a double meaning (IBAN *or* account number) in some cases? And in which cases? If there are different kinds of values in the `accountNumber` field we should mention that in a comment in `index.ts`.

As far as I understand the intention of this function - you need to check different validity fields depending on the "mode" (IBAN or account number), right? If that's the case, maybe we could move the check into a named function, e.g. `userWantsAccountNumber` or `userWantsIban` or add some comments here.

Reply:
The accountNumber/bankCode items are only used as the models for the 2 fields, because we only care about submitting the IBAN/BIC.

Theoretically they could be stripped out of the store and made refs, and the validity checks would be done on those directly in the BankFields component. If (when?) we implement an IBAN calculator these will get removed.